### PR TITLE
Add copyright and checks

### LINKED ISSUE #2: Decide on copyright for code
We have a licence (Apache2 👍), but we also need the code to be copyrighted.

My suggestion would be that we (C) to CERN as the "mother lab" for the project. That should be ok for all HEP contributors.

Then all source files should have:

```
// SPDX-FileCopyrightText: 2020 CERN
// SPDX-License-Identifier: Apache-2.0 
```

We should also write a contribution guide and this should be in.

I also found that we can have a GitHub action that would check compliance, see <https://reuse.software/>.

### PR DESCRIPTION
**Pull request by graeme-a-stewart**
Add (C) to CERN for source files and Apache-2.0 license in SPDX format.
Add (C) for Resuse action which is GPL3+.

All licenses in the project are also stored in the `LICENSES` sub-directory. For the main Apache-2.0 license this is implemented as a softlink to `./LICENSE`.

Added reuse action to ensure that license and (C) are present in all files (Reuse guidelines are strict about *all* files!).

README updated with a short note about (C) and licensing.

Added AUTHORS file to ensure that all contributors and their institutes are acknowledged.

Closes #2

### CODE REVIEW

**Comment by graeme-a-stewart** [comment]
Changed documentation license to CC-BY-4.0 and used an HTML comment for this, rather than a markdown codeblock.

File: AUTHORS.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+SPDX-FileCopyrightText: 2020 CERN
+SPDX-License-Identifier: CC-BY-4.0
+-->
+
+# Authors
+
+The following people have contributed to the AdePT project (in alphabetical order):
+
+- Andrei Gheata, CERN
+- Benjamin Morgan, University of Warwick
```
Comment (drbenmorgan) [inline AUTHORS.md:11]:
```suggestion
- Ben Morgan, University of Warwick
```

I tend to go by the short version!

**Review by drbenmorgan — CHANGES_REQUESTED** [review]

File: AUTHORS.md (hunk 1 above)
Comment (graeme-a-stewart) [reply AUTHORS.md]: Absolutely!

**Review by drbenmorgan — APPROVED** [review]
Thanks @graeme-a-stewart !

**Review by agheata — APPROVED** [review]

**Pull request merged**